examples/plot_srm_alignment.py

```diff
@@ -0,0 +1,211 @@
+
+imgs, df, mask_img = fetch_ibc_subjects_contrasts(
+    ["sub-01", "sub-02", "sub-04", "sub-05", "sub-06", "sub-07"]
+)
+
+###############################################################################
+# Definine a masker
+# -----------------
+# We define a nilearn masker that will be used to handle relevant data.
+#   For more information, visit :
+#   'http://nilearn.github.io/manipulating_images/masker_objects.html'
+#
+
+from nilearn.maskers import NiftiMasker
+
+masker = NiftiMasker(mask_img=mask_img).fit()
+
+###############################################################################
+# Prepare the data
+# ----------------
+# For each subject, we will use two series of contrasts acquired during
+# two independent sessions with a different phase encoding:
+# Antero-posterior(AP) or Postero-anterior(PA).
+#
+
+
+# To infer a template for subjects sub-01 to sub-06 for both AP and PA data,
+# we make a list of 4D niimgs from our list of list of files containing 3D images
+
+from nilearn.image import concat_imgs
+
+template_train = []
+for i in range(6):
+    template_train.append(concat_imgs(imgs[i]))
+target_train = df[df.subject == "sub-07"][df.acquisition == "ap"].path.values
+
+# For subject sub-07, we split it in two folds:
+#   - target train: sub-07 AP contrasts, used to learn alignment to template
+#   - target test: sub-07 PA contrasts, used as a ground truth to score predictions
+# We make a single 4D Niimg from our list of 3D filenames
+
+target_train = concat_imgs(target_train)
+target_train_data = masker.transform(target_train)
+target_test = df[df.subject == "sub-07"][df.acquisition == "pa"].path.values
+
+###############################################################################
+# Compute a baseline (average of subjects)
+# ----------------------------------------
+# We create an image with as many contrasts as any subject representing for
+# each contrast the average of all train subjects maps.
+#
+
+import numpy as np
+
+masked_imgs = [masker.transform(img) for img in template_train]
+average_img = np.mean(masked_imgs[:-1], axis=0)
+average_subject = masker.inverse_transform(average_img)
+
+###############################################################################
+# Create a template from the training subjects.
+# ---------------------------------------------
+# We define an estimator using the class TemplateAlignment:
+#   * We align the whole brain through 'multiple' local alignments.
+#   * These alignments are calculated on a parcellation of the brain in 150 pieces,
+#     this parcellation creates group of functionnally similar voxels.
+#   * The template is created iteratively, aligning all subjects data into a
+#     common space, from which the template is inferred and aligning again to this
+#     new template space.
+#
+
+from nilearn.image import index_img
+from fastsrm.identifiable_srm import IdentifiableFastSRM
+
+
+train_index = range(53)
```

Comment:
I don't think we've carried over a comment about this being 53 contrasts, so the index seems pretty mysterious to the reader